module/metrics/server.go

```diff
@@ -3,20 +3,31 @@ package metrics
 import (
 	"context"
 	"errors"
+	"net"
 	"net/http"
 	"strconv"
 	"time"
 
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
 )
 
+// metricsServerShutdownTimeout is the time to wait for the server to shut down gracefully
+const metricsServerShutdownTimeout = 5 * time.Second
+
 // Server is the http server that will be serving the /metrics request for prometheus
 type Server struct {
 	server *http.Server
 	log    zerolog.Logger
+
+	startupCompleted chan struct{}
 }
 
+var _ component.Component = (*Server)(nil)
```

Comment:
I think this should use a component manager instead of custom lifecycle management. There's a similar implementation for using a component manager for a server in the access rpc engine.

https://github.com/onflow/flow-go/blob/master/engine/access/rpc/engine.go#L125-L134